File: app/src/main/java/com/mithilakshar/mithilapanchang/holidayadapter.java
package com.mithilakshar.mithilapanchang;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.squareup.picasso.Picasso;

import java.util.ArrayList;

public class holidayadapter extends RecyclerView.Adapter<holidayviewholder> {

    ArrayList<holidaydatamodel> datalist;

    public holidayadapter(ArrayList<holidaydatamodel> datalist) {
        this.datalist = datalist;
    }

    @NonNull
    @Override
    public holidayviewholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view=LayoutInflater.from(parent.getContext()).inflate(R.layout.holidayitem,parent,false);
        return new holidayviewholder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull holidayviewholder holder, int position) {

        holder.holidayName.setText(datalist.get(position).getHolidayName());
        holder.holidayDesc.setText(datalist.get(position).getHolidayDesc());
        Picasso.get().load(datalist.get(position).getImageUrl()).into(holder.holidayImage);


    }

    @Override
    public int getItemCount() {
        return datalist.size();
    }
}

File: app/src/main/java/com/mithilakshar/mithilapanchang/holidayadapter.kt
package com.mithilakshar.mithilapanchang

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.squareup.picasso.Picasso

class holidayadapter(var datalist: ArrayList<holidaydatamodel?>) :
    RecyclerView.Adapter<holidayviewholder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): holidayviewholder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.holidayitem, parent, false)
        return holidayviewholder(view)
    }

    override fun onBindViewHolder(holder: holidayviewholder, position: Int) {


    }

    override fun getItemCount(): Int {
        return datalist.size
    }
}
